Source/SelfService/Web/microservice/microserviceView/configuration/environmentVariablesSection/environmentVariableSection.tsx

```diff
@@ -0,0 +1,288 @@
+    const [restartInfoBoxIsOpen, setRestartInfoBoxIsOpen] = useState(false);
+
+    const columns: GridColDef<EnvironmentVariableTableRow>[] = [
+        {
+            field: 'name',
+            headerName: 'Name',
+            width: 330,
+            editable: true
+        },
+        {
+            field: 'value',
+            headerName: 'Value',
+            width: 330,
+            editable: true
+        },
+        {
+            field: 'isSecret',
+            headerName: 'Secret',
+            type: 'singleSelect',
+            valueOptions: [{ value: true, label: 'Yes' }, { value: false, label: 'No' }],
+            editable: true,
+            renderCell: ({ value }) => (
+                // TODO: Needs improvement
+                <Button
+                    variant='text'
+                    label={value ? 'Yes' : 'No'}
+                    sx={{ color: 'text.primary' }}
+                    endWithIcon={<ArrowDropDown />}
+                />
+            ),
+            width: 90
+        }
+    ];
+
+    useEffect(() => {
+        fetchAndUpdateEnvVariableList()
+            .catch(console.error);
+    }, []);
+
+    const fetchAndUpdateEnvVariableList = async () => {
+        const result = await getEnvironmentVariables(applicationId, environment, microserviceId);
+
+        result.data ?
+            createDataTableObj(result.data) :
+            enqueueSnackbar('Could not fetch environment variables.', { variant: 'error' });
+    };
+
+    const createDataTableObj = (envVariables: InputEnvironmentVariable[]): void => {
+        const rows = envVariables.map(envVariable => {
+            return {
+                id: envVariable.name,
+                name: envVariable.name,
+                value: envVariable.value,
+                isSecret: envVariable.isSecret
+            } as EnvironmentVariableTableRow;
+        });
+
+        setEnvVariableTableRows(rows);
+    };
+
+    const validateEnvVariable = (envVariable: EnvironmentVariableTableRow): boolean => {
+        if (envVariable.name === '' || envVariable.value === '') {
+            enqueueSnackbar('You cant have an empty name or value.', { variant: 'error' });
+            return false;
+        }
+
+        if (envVariableTableRows.some(row => row.name === envVariable.name && row.id !== envVariable.id)) {
+            enqueueSnackbar('You cant have duplicate names.', { variant: 'error' });
+            return false;
+        }
+
+        return true;
+    };
+
+    const handleInvalidValues = (id: GridRowId) => {
+        setRowMode({
+            ...rowMode,
+            [id]: { mode: GridRowModes.View, ignoreModifications: true },
+        });
+
+        const editedRow = envVariableTableRows.find(row => row.id === id);
+
+        if (editedRow!.isNew) {
+            setEnvVariableTableRows(envVariableTableRows.filter(row => row.id !== id));
+        }
+
+        // TODO: Check if there are any changes before updating
```

Comment:
Aha, this looks like it's related to a previous comment I made earlier. So, don't attempt to save if the values haven't changed, right? Same comment as above, we can solve this in a follow-up PR or look at this together, perhaps?